src/status_im/contexts/profile/backup_recovery_phrase/view.cljs

```diff
@@ -2,123 +2,167 @@
-        [rn/view {:style {:flex 1}}
-         [quo/page-nav
-          {:icon-name           :i/close
-           :on-press            #(rf/dispatch [:navigate-back])
-           :accessibility-label :top-bar}]
-         [quo/page-top
-          {:title            (i18n/label :t/backup-recovery-phrase)
-           :description      :text
-           :description-text (i18n/label :t/backup-recovery-phrase-description)
-           :container-style  {:padding-bottom 8}}]
-         [rn/view {:style (style/seed-phrase-container theme)}
-          (when (pos? (count @seed-phrase))
-            [:<>
-             [words-column
-              {:words       @seed-phrase
-               :first-half? true}]
-             [rn/view {:style (style/separator theme)}]
-             [words-column
-              {:words       @seed-phrase
-               :first-half? false}]])
-          (when-not @revealed?
-            [rn/view {:style style/blur-container}
-             [quo/blur (style/blur theme)]])]
-         (when-not @revealed?
-           [rn/view
-            {:style {:padding-horizontal 20
-                     :padding-top        20}}
-            [quo/text
-             {:weight :semi-bold
-              :style  {:margin-bottom 8}}
-             (i18n/label :t/how-to-backup)]
-            [rn/flat-list
-             {:data           step-labels
-              :render-fn      step-item
-              :render-data    {:checked?            checked?
-                               :customization-color customization-color}
-              :scroll-enabled false}]])
-         (if @revealed?
-           [rn/view {:style style/slide-button}
-            [quo/bottom-actions
-             {:actions          :one-action
-              :button-one-label (i18n/label :t/i-have-written)
-              :button-one-props {:customization-color customization-color
-                                 :on-press            #(on-success (security/mask-data @seed-phrase))}}]
-            [quo/text
-             {:size  :paragraph-2
-              :style (style/description-text theme)}
-             (i18n/label :t/next-you-will)]]
-           [quo/bottom-actions
-            {:actions          :one-action
-             :button-one-label (i18n/label :t/reveal-phrase)
-             :button-one-props {:disabled?           (some false? (vals @checked?))
-                                :customization-color customization-color
-                                :on-press            #(reset! revealed? true)}
-             :container-style  style/slide-button}])]))))
+  (let [{:keys             [on-success masked-seed-phrase shell? back-button?]
+         initial-revealed? :revealed?} (quo.context/use-screen-params)
+        [revealed? set-revealed]       (rn/use-state initial-revealed?)
+        reveal-seed-phrase             (rn/use-callback #(set-revealed true))
+        [seed-phrase set-seed-phrase]  (rn/use-state (or (some-> masked-seed-phrase
```

Comment:
An interesting abstraction to think about in the future is to have a special hook that abstracts away the manual task of persisting the text masked, which is more error prone. Something like `use-masked-text`. Also the hook would always return a masked text so there would be less of a risk of pass around an unmasked text around.

Reply:
It'd be interesting to introduce something like that.